Makefile

```diff
@@ -1,24 +1,25 @@
 S3_CP_ARGS=aws s3 cp --acl public-read
+RELEASE_VERSION?=
+
+S3_BUCKET_PATH=s3://observe/cloudformation/
+S3_EXISTING_VERSIONS=$(shell aws s3 ls $(S3_BUCKET_PATH) | grep $(RELEASE_VERSION) | wc -l)
 
 .PHONY: cloudformation
 cloudformation:
-	$(S3_CP_ARGS) templates/collection.yaml s3://observeinc/cloudformation/collection-`semtag final -s minor -o`.yaml
-	$(S3_CP_ARGS) templates/collection.yaml s3://observeinc/cloudformation/collection-latest.yaml
-	$(S3_CP_ARGS) templates/config.yaml s3://observeinc/cloudformation/config-`semtag final -s minor -o`.yaml
-	$(S3_CP_ARGS) templates/config.yaml s3://observeinc/cloudformation/config-latest.yaml
-	$(S3_CP_ARGS) templates/cwmetricsstream.yaml s3://observeinc/cloudformation/cwmetricsstream-`semtag final -s minor -o`.yaml
-	$(S3_CP_ARGS) templates/cwmetricsstream.yaml s3://observeinc/cloudformation/cwmetricsstream-latest.yaml
-	$(S3_CP_ARGS) templates/quicksetup.yaml s3://observeinc/cloudformation/quicksetup-`semtag final -s minor -o`.yaml
-	$(S3_CP_ARGS) templates/quicksetup.yaml s3://observeinc/cloudformation/quicksetup-latest.yaml
-	$(S3_CP_ARGS) templates/vpc.yaml s3://observeinc/cloudformation/vpc-`semtag final -s minor -o`.yaml
-	$(S3_CP_ARGS) templates/vpc.yaml s3://observeinc/cloudformation/vpc-latest.yaml
-	$(S3_CP_ARGS) templates/controltower.yaml s3://observeinc/cloudformation/controltower-`semtag final -s minor -o`.yaml
-	$(S3_CP_ARGS) templates/controltower.yaml s3://observeinc/cloudformation/controltower-latest.yaml
-
-.PHONY: changelog
-changelog:
-	git-chglog -o CHANGELOG.md --next-tag `semtag final -s minor -o`
+	for file in templates/*.yaml ; do \
+		$(S3_CP_ARGS) $$file $(S3_BUCKET_PATH)`basename $$file .yaml`-$(RELEASE_VERSION).yaml ; \
```

Comment:
`$()` is generally preferred to backticks:

http://mywiki.wooledge.org/BashFAQ/082

Even with the need to escape the dollar this is clearer:

```
$$(basename $$file .yaml)
```

Reply:
resolved in [e93dc8f](https://github.com/observeinc/cloudformation-aws-collection/pull/35/commits/e93dc8f8fbfd74a79452f2fa9adadc7dadd41c64)